src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClient.h

```diff
@@ -31,22 +31,19 @@ namespace client
     class AwsSmithyClientT : public AwsSmithyClientBase
     {
     public:
-        AwsSmithyClientT(Aws::Client::ClientConfiguration& clientConfig, const Aws::String& serviceName,
+        AwsSmithyClientT(const ServiceClientConfigurationT& clientConfig, const Aws::String& serviceName,
             const std::shared_ptr<Aws::Http::HttpClient>& httpClient,
             const std::shared_ptr<Aws::Client::AWSErrorMarshaller>& errorMarshaller,
-            const ServiceClientConfigurationT& m_client_config,
             const std::shared_ptr<EndpointProviderT> endpointProvider,
-            const std::shared_ptr<ServiceAuthSchemeResolverT>& m_auth_scheme_resolver,
-            const Aws::UnorderedMap<Aws::String, AuthSchemesVariantT>& m_auth_schemes)
+            const std::shared_ptr<ServiceAuthSchemeResolverT>& authSchemeResolver,
```

Comment:
why not const shared_ptr<const T> p . The interface should clearly indicate the intention. Non const means I can change the contests inside the class